### PR TITLE
Add support for init operations

### PR DESCRIPTION
**Pull request by Tabrizian**

### CODE REVIEW

File: src/tensorflow.cc
Comment (Tabrizian) [reply src/tensorflow.cc:1010]: I think it should be fine. I don't think they have the ability to inject code here. TF will just check whether the name exists in the list of operators or not.

File: README.md
Comment (tanmayv25) [inline README.md:232]: I was wondering if it would be better to have `init_ops.json` per version of the model? 
A user might want to initialize the variables using different ops for different model versions.

File: src/tensorflow_backend_tf.cc
Comment (tanmayv25) [inline src/tensorflow_backend_tf.cc:1]: Just confirming, you copied the file from TensorFlow repository directly here, right?

**Review by tanmayv25 — CHANGES_REQUESTED** [review]

**Comment by tanmayv25** [comment]
Also, you should wait for your upstream MRs to be merged and frozen before merging this and the server PRs.

File: README.md
Comment (Tabrizian) [reply README.md:232]: How about having a hierarchy? If there is a `init_ops.json` inside the version directory it will be used. Otherwise it will lookup the model folder?

File: src/tensorflow_backend_tf.cc
Comment (Tabrizian) [reply src/tensorflow_backend_tf.cc:1]: Correct.

**Comment by Tabrizian** [comment]
I think we should probably wait on freezing these PRs until the TF container is frozen and this PR can directly go to the 22.07 branch.

**Comment by tanmayv25** [comment]
Exactly!

File: README.md
Comment (tanmayv25) [reply README.md:232]: Sounds good to me.

File: README.md
Comment (tanmayv25) [inline README.md:221]: Should update the documentation to match the current behavior. 
Say that json file can be provided per version directory or model directory. Talk about which one will take precedence and then give some examples.

**Review by tanmayv25 — COMMENTED** [review]
Looks good. Not approving to avoid accidental merge.

**Review by tanmayv25 — APPROVED** [review]

File: src/tensorflow_backend_tf.h
Comment (jbkyang-nvi) [inline src/tensorflow_backend_tf.h:1]: should this be 2019-2022?

Comment (tanmayv25) [reply src/tensorflow_backend_tf.h:1]: I think now this is tracking TF2 repo wrapper?

Comment (jbkyang-nvi) [reply src/tensorflow_backend_tf.h:1]: yes, does that mean the tf2 repo copyright years are wrong?

Comment (tanmayv25) [reply src/tensorflow_backend_tf.h:1]: May be the wrapper for TF2 was written in 2020 and not in 2019. I would still prefer this to be an exact copy of the TF2 upstream.

Comment (jbkyang-nvi) [reply src/tensorflow_backend_tf.h:1]: I checked and it has not been updated...I think we should at least fix the github one for this release and update the same on the tensorflow branch?

Comment (tanmayv25) [inline src/tensorflow_backend_tf.h:1]:
```suggestion
// Copyright 2019-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```

**Pull request merged**
